libr/include/r_types_overflow.h

```diff
@@ -0,0 +1,53 @@
+#ifndef R_TYPES_OVERFLOW_H
+#define R_TYPES_OVERFLOW_H
+
+/* preventive math overflow checks */
+#if !defined(SZT_ADD_OVFCHK)
+#define SZT_ADD_OVFCHK(x,y) ((SIZE_MAX - (x)) < (y))
+#endif
+
+#if 0
```

Comment:
just use comments instead of `#if 0` if you want to explain something.

Reply:
this if0 code is from the stackoverflow page you shared, it's there for reference, basically because not all conditions are done, and not all cases are handled in the suite. but i wanted to push the pr before falling asleep to get some feedback and maybe catch an issue here or there

Reply:
yeah it is ok to keep the code inside the `#if 0`. I'm just saying, just use comments instead of using `#if 0` to comment stuff.